### ISSUE TITLE
Add Boost.Beast experimental streaming parser

### ISSUE THREAD

**Issue by mloskot** [miloyip/nativejson-benchmark#118]
The parser implementation is available from Boost.Beast's demo project here
https://github.com/vinniefalco/BeastLounge/tree/develop/include/boost/beast/_experimental/json

/cc @vinniefalco

**Comment by vinniefalco** [comment]
Yeah, I would love to see a benchmark **BUT** full disclosure, I have spent exactly zero time optimizing this code (LOL)

**Comment by mloskot** [comment]
@vinniefalco I wouldn't worry about it. The benchmark compares what is available 'on the market' as it is.
AFAIK, there has been just first round of implementation of the Ciere's Spirit-based parser, so it has not been optimised either.
Still, it is valuable to have it compared, for both, its developers and its users.

**Comment by vinniefalco** [comment]
The streaming parser is separate from the non-streaming parser now. They have different performance characteristics.
